apps/preview/nuxt/pages/showcases/Combobox/ComboboxBasic.vue

```diff
@@ -0,0 +1,263 @@
+      :wrapper-class="{
+        '!bg-disabled-100 !ring-disabled-300 !hover:ring-disabled-300 !ring-1': isDisabled,
+      }"
+      @blur="handleBlur"
+      @focus="isValid = undefined"
+      @click="toggle()"
+      @keydown="handleInputKeyDown"
+      ><template #suffix>
+        <SfIconExpandMore
+          :class="[
+            'ml-auto text-neutral-500 transition-transform ease-in-out duration-300',
+            {
+              'rotate-180': isOpen,
+              '!text-disabled-500 cursor-not-allowed': isDisabled,
+            },
+          ]"
+          @click="!isDisabled && toggle()"
+        />
+      </template>
+    </SfInput>
+    <div v-if="isOpen" ref="floatingRef" :style="style" class="left-0 right-0">
+      <div
+        v-if="isLoadingSnippets"
+        class="flex items-center justify-center w-full h-20 py-2 bg-white border border-solid rounded-md border-neutral-100 drop-shadow-md"
+      >
+        <SfLoaderCircular />
+      </div>
+      <ul
+        v-else
+        :id="listId"
+        ref="dropdownRef"
+        role="listbox"
+        aria-label="Country list"
+        class="py-2 bg-white border border-solid rounded-md border-neutral-100 drop-shadow-md"
+      >
+        <template v-if="snippets.length > 0">
+          <li v-for="option in snippets" :key="option.value">
+            <SfListItem
+              :id="`${listId}-${option.value}`"
+              tag="button"
+              type="button"
+              class="flex justify-start"
+              :aria-selected="option.value === inputModel"
+              @click="selectOption(option.label)"
+              @keydown.enter.space.prevent="selectOption(option.label)"
+              @keydown.esc="handleFocusInput"
+            >
+              <p class="text-left">
+                <span>
+                  {{ option.label }}
+                </span>
+              </p>
+            </SfListItem>
+          </li>
+        </template>
+        <SfListItem v-else-if="inputModel" class-name="flex justify-start" aria-label="No options">
+          <p class="text-left">
+            <span>No options</span>
+          </p>
+        </SfListItem>
+        <template v-else>
+          <li v-for="option in options" :key="option.value">
+            <SfListItem
+              :id="`${listId}-${option.value}`"
+              tag="button"
+              type="button"
+              class="flex justify-start"
+              :aria-selected="option.value === inputModel"
+              @click="selectOption(option.label)"
+              @keydown.enter.space.prevent="selectOption(option.label)"
+              @keydown.esc="handleFocusInput"
+            >
+              <p class="text-left">
+                <span>{{ option.label }}</span>
+              </p>
+            </SfListItem>
+          </li>
+        </template>
+      </ul>
+    </div>
+  </div>
+  <p v-if="!isDisabled && isValid === false && !isOpen" class="text-negative-700 typography-text-sm font-medium mt-0.5">
```

Comment:
Suggestion: Let's clear error state only when state of the field changes, not immediately after combobox opens.
```suggestion
  <p v-if="!isDisabled && isValid === false" class="text-negative-700 typography-text-sm font-medium mt-0.5">
```

Reply:
Ok, removed. I added it because when the focus moved to dropdown error appeared (but now the message is hidden under dropdown so it doesn't break anything)